src/EditorFeatures/CSharpTest/Diagnostics/GenerateMethod/GenerateMethodTests.cs

```diff
@@ -2757,17 +2757,44 @@ private void Test()
         public async Task TestGenerateMethodWithMethodChaining()
         {
             await TestAsync(
-@"using System ; using System . Collections . Generic ; using System . Linq ; using System . Threading . Tasks ; class Program { static void Main ( string [ ] args ) { bool x = await [|Foo|] ( ) . ConfigureAwait ( false ) ; } } ",
-@"using System ; using System . Collections . Generic ; using System . Linq ; using System . Threading . Tasks ; class Program { static void Main ( string [ ] args ) { bool x = await Foo ( ) . ConfigureAwait ( false ) ; } private static Task < bool > Foo ( ) { throw new NotImplementedException ( ) ; } } ");
+@"using System ;
+using System . Collections . Generic ;
+using System . Linq ;
+using System . Threading . Tasks ;
+class Program {
+    static void Main ( string [ ] args ) {
+        bool x = await [|Foo|] ( ) . ConfigureAwait ( false ) ;
+    }
+}",
+@"using System ;
+using System . Collections . Generic ;
+using System . Linq ;
+using System . Threading . Tasks ;
+class Program {
+    static void Main ( string [ ] args ) {
+        bool x = await Foo ( ) . ConfigureAwait ( false ) ;
+    }
+    private static Task < bool > Foo ( ) { throw new NotImplementedException ( ) ; } } ");
         }
 
         [WorkItem(643, "https://github.com/dotnet/roslyn/issues/643")]
         [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateMethod)]
         public async Task TestGenerateMethodWithMethodChaining2()
         {
             await TestAsync(
-@"using System ; using System . Threading . Tasks ; class C { static async void T ( ) { bool x = await [|M|] ( ) . ContinueWith ( a => { return true ; } ) . ContinueWith ( a => { return false ; } ) ; } } ",
-@"using System ; using System . Threading . Tasks ; class C { static async void T ( ) { bool x = await M ( ) . ContinueWith ( a => { return true ; } ) . ContinueWith ( a => { return false ; } ) ; } private static Task < bool > M ( ) { throw new NotImplementedException ( ) ; } } ");
+@"using System ;
+using System . Threading . Tasks ;
+class C {
+    static async void T ( ) {
+        bool x = await [|M|] ( ) . ContinueWith ( a => { return true ; } ) . ContinueWith ( a => { return false ; } ) ;
+    }
+} ",
+@"using System ;
+using System . Threading . Tasks ;
+class C {
+    static async void T ( ) {
+        bool x = await M ( ) . ContinueWith ( a => { return true ; } ) . ContinueWith ( a => { return false ; } ) ; }
+        private static Task<object> M ( ) { throw new NotImplementedException ( ) ; } } ");
```

Comment:
Yes.  The inference of 'bool' was specious.  It basically was saying the equivalent of: i have "bool b = x.y.z", ergo 'x' and 'y' are bool".  This simply is not an accurate or appropriate inference to make.